server/szurubooru/func/posts.py

```diff
@@ -967,3 +972,81 @@ def search_by_image(image_content: bytes) -> List[Tuple[float, model.Post]]:
         ]
     else:
         return []
+
+
+PoolPostsAround = namedtuple('PoolPostsAround', 'pool first_post prev_post next_post last_post')
+
+def get_pool_posts_around(post: model.Post) -> List[PoolPostsAround]:
+    around = dict()
+    pool_ids = set()
+    post_ids = set()
+
+    dbquery = """
+    SELECT around.ord, around.pool_id, around.post_id, around.delta
+    FROM pool_post pp,
+        LATERAL get_pool_posts_around(pp.pool_id, pp.post_id) around
+    WHERE pp.post_id = :post_id;
+    """
+
+    for order, pool_id, post_id, delta in db.session.execute(dbquery, {"post_id": post.post_id}):
```

Comment:
When using SQLAlchemy, you should just capture the `model.Pool` and `model.Post` objects directly instead of working with the IDs